modules/akka-cache/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/akka/cache/CacheActor.scala

```diff
@@ -98,6 +118,16 @@ object CacheActor {
       */
     final case class Put[V](key: String, value: V) extends Protocol
 
+    /**
+      * Put if absent message. When received, the actor set its in-memory value to the provided ''value''
+      * ONLY if no previous value exists for the provided ''key''.
+      *
+      * @param key   the key for which the provided ''value'' is going to be added if no value already exists
+      * @param value the value to be added if no value already exists
+      * @tparam V the generic type of the value
+      */
+    final case class PutIfAbsent[V](key: String, value: () => V) extends Protocol
```

Comment:
the value needs to be eagerly evaluated, otherwise the entire function and its context will have to be serialized when crossing the boundaries of an actor system; using a function is strongly discouraged. Please update to `value: V`